update #6380 with addressed some review feedback

poszu opened the pull request:
Updates the #6380 with some fixed review remarks

:bulb: I recommend looking commit by commit - changes are separated by commits.

poszu committed:
removed system.ValidationRequestNew

poszu committed:
Use zaptest in tests

poszu committed:
refactor getting athena library path

jellonek commented on Makefile-libs.Inc:
```diff
@@ -119,8 +119,8 @@ $(BINDIR_ATHENA_SETUP_LIBS): $(BIN_DIR).athena.version.$(ATHENA_SETUP_REV)
 ATHENA_LIB_RENAMED := libathenavmwrapper$(suffix $(ATHENA_LIB))
-
-$(BIN_DIR)$(ATHENA_LIB_RENAMED): $(BIN_DIR)$(ATHENA_LIB)
+ATHENA_LIB_PATH := $(BIN_DIR)$(ATHENA_LIB_RENAMED)
```
That may confuse someone as most of the time env variables with PATH suffix contain list of directories to search for an entry.
IMO better name would be ATHENA_LIB_LOCATION.

poszu committed:
infallible address calculation

poszu committed:
don't use precompiled programs for UTs

poszu committed:
fix missing vm destroying

poszu commented on Makefile:
```diff
@@ -103,7 +103,7 @@ clear-test-cache:
 
 test: get-libs
-	@$(ULIMIT) CGO_LDFLAGS="$(CGO_TEST_LDFLAGS)" gotestsum -- -race -p 1 -timeout 8m $(UNIT_TESTS)
+	@$(ULIMIT) ATHENA_LIB_PATH="$(ATHENA_LIB_PATH)" CGO_LDFLAGS="$(CGO_TEST_LDFLAGS)" gotestsum -- -timeout 8m $(UNIT_TESTS)
```
Revert `-race -p 1` deletion

poszu committed:
fix test target

poszu committed:
fix wallet template address in athena preset

poszu committed:
include athena library in node Docker image

poszu committed:
fix devnet-athena-8 genesis data

because we cannot change it for a living network.

Added the wallet template account from code (not in the genesis data).

poszu committed:
fix fixing wallet template address

lrettig reviewed (approved):
lgtm
